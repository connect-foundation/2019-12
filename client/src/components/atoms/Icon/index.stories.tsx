import React from 'react';
import { text } from '@storybook/addon-knobs';

import Icon from '.';
import ExternalLinkSymbol from '../../../assets/img/external-link-symbol.svg';

export default {
  title: 'Atoms / Icon',
};

export const index: React.FC = () => (
  <div style={{ padding: '1rem', backgroundColor: 'gray' }}>
    <Icon
      alt={text('alt', 'External Link Icon')}
      height={text('height', '2rem')}
      src={ExternalLinkSymbol}
    />
  </div>
);

export const circular: React.FC = () => (
  <div style={{ padding: '1rem', backgroundColor: 'gray' }}>
    <Icon
      alt={text('alt', 'External Link Icon')}
      height={text('height', '2rem')}
      src={ExternalLinkSymbol}
      circular={true}
    />
  </div>
);
