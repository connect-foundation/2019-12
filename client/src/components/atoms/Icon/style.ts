import styled from 'styled-components';
import { prop, ifProp } from 'styled-tools';

export const Img = styled.img`
  height: ${props => props.height};
  border-radius: ${ifProp('circular', '50%', '')};
`;
