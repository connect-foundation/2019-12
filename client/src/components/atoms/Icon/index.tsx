import React from 'react';

import * as S from './style';

export interface IconProps {
  /** 접근성 대응 */
  alt: string;
  /** 크기 */
  height?: string;
  /** 아이콘 소스 */
  src: any;
  /** custom style */
  style?: object;
  /** circular img */
  circular?: boolean;
}

function Icon({ height = '2rem', ...props }: IconProps): React.ReactElement {
  return <S.Img height={height} {...props} />;
}

export default Icon;
