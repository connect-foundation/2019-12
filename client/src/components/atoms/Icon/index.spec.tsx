import React from 'react';
import { mount } from 'enzyme';
import Icon from '.';
import ExternalLinkSymbol from '../../../assets/img/external-link-symbol.svg';

describe('Atom / Divider', () => {
  it('[SNAPSHOT] 렌더링', () => {
    // given
    const wrapper = mount(<Icon alt="string" src={ExternalLinkSymbol} />);

    // then
    expect(wrapper).toMatchSnapshot();
  });
});
