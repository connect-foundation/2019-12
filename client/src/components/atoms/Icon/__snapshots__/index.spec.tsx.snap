// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom / Icon [SNAPSHOT] Circular 렌더링 1`] = `
.c0 {
  height: 2rem;
  border-radius: 50%;
}

<Icon
  alt="External Link Icon"
  circular={true}
  height="2rem"
  src="external-link-symbol.svg"
>
  <styled.img
    alt="External Link Icon"
    circular={true}
    height="2rem"
    src="external-link-symbol.svg"
  >
    <StyledComponent
      alt="External Link Icon"
      circular={true}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  height: ",
              [Function],
              ";
  border-radius: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.img",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "img",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      height="2rem"
      src="external-link-symbol.svg"
    >
      <img
        alt="External Link Icon"
        className="c0"
        height="2rem"
        src="external-link-symbol.svg"
      />
    </StyledComponent>
  </styled.img>
</Icon>
`;

exports[`Atom / Icon [SNAPSHOT] 렌더링 1`] = `
.c0 {
  height: 2rem;
}

<Icon
  alt="string"
  src="external-link-symbol.svg"
>
  <styled.img
    alt="string"
    height="2rem"
    src="external-link-symbol.svg"
  >
    <StyledComponent
      alt="string"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  height: ",
              [Function],
              ";
  border-radius: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.img",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "img",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      height="2rem"
      src="external-link-symbol.svg"
    >
      <img
        alt="string"
        className="c0"
        height="2rem"
        src="external-link-symbol.svg"
      />
    </StyledComponent>
  </styled.img>
</Icon>
`;
